src/naip_cnn/sampling.py

```diff
@@ -0,0 +1,258 @@
+"""
+This module contains functions for sampling data from local rasters and Earth Engine 
+images.
+"""
+
+from __future__ import annotations
+
+import io
+import math
+
+import ee
+import geopandas as gpd
+import numpy as np
+import pandas as pd
+import rasterio
+from joblib import Parallel, delayed
+from PIL import Image
+
+
+def _create_point_grid(*, bounds, spacing: float, crs: str) -> gpd.GeoDataFrame:

```

Comment:
Yeah, that was my thinking - do you think we might need to use it elsewhere in the project?

I'm a fan of the required kwargs too! I have a bad habit of calling my own functions with positional args, and got bit a few times when I changed the function signature without updating the calls.

Reply:
> Yeah, that was my thinking - do you think we might need to use it elsewhere in the project?

Only in the sense that it's a useful function for other tasks (even outside this package).  I often struggle with where to keep these utility type functions that I keep writing over and over for different projects.  Not suggesting that you do this now, but do have thoughts on creating a separate "swiss army knife" package or what would be best practices on when you'd need this function again elsewhere?

Reply:
This is something I definitely run into as well, and so far I've always ended up re-implementing these utility functions again or copy-pasting between projects - definitely not the most efficient solution! I like the idea of a "swiss army knife" package, but I've never actually gone to the trouble of building one.

One of the reasons I've avoided it is that it would be difficult to integrate into public projects, e.g. we wouldn't want to include something like that it in sknnr because it would bloat the dependencies, and using it in published research would mean the package needs to be published as well. But there's no doubt it would be useful for internal projects, and if nothing else it would at least get those functions in one place if they need to be copied into other projects. I'm totally open to the idea of collaborating on a general purpose, geospatial data science package!